### PR TITLE
Drop TS dependencies for Trinity

### PR DESCRIPTION
**Pull request by mvdbeek**
Remove trinity from testing blacklist, update README and use
detect_errors="aggressive" throughout.

### CODE REVIEW

File: tools/trinity/run_de_analysis.xml [hunk 1]
```diff
@@ -5,15 +5,13 @@
     </macros>
     <expand macro="requirements">
         <!-- odering is crucial, otherwise R will override the ENV variables from deseq2 -->
```
Comment (nsoranzo) [inline tools/trinity/run_de_analysis.xml]: This comment can be removed.

File: tools/trinity/README.rst [hunk 2]
```diff
@@ -6,13 +6,25 @@ As described on the official Trinity website
 trinity requires a large amount of memory to perform the assembly: "roughly
 ~1G of RAM per 1M reads to be assembled"
 
-By default, this tool is configured to limit the memory consumption to 30G.
+By default, this tool is configured to limit the memory consumption to 1G.
 You might need to lower this limit if the machine(s) executing the jobs have less memory available.
 If you have a lot of reads to assemble and a machine with enough memory, you can increase it.
-In both cases, you can edit the TRINITY_MEM_OPTIONS in the file:
+In both cases, you can set the TRINITY_MEM_OPTIONS environmental variable in the destination section of the job_conf.xml file:
```
Comment (nsoranzo) [inline tools/trinity/README.rst]: s/environmental/environment/

File: tools/trinity/tool_dependencies.xml [hunk 3]
```diff
@@ -1,30 +0,0 @@
...
-        <repository name="package_express_1_5_1" owner="iuc" />
-    </package>
-    <package name="bioconductor-limma" version="3.28.10">
-        <repository name="package_bioconductor_limma_3_28_10" owner="iuc" />
-    </package>
-    <package name="deseq2" version="1.10.0">
-       <repository name="package_deseq2_1_10_0" owner="iuc" />
-    </package>
-    <set_environment version="1.0">
-        <environment_variable name="TRINITY_MEM_OPTIONS" action="set_to">'--max_memory 30G --bflyHeapSpaceMax 30G'</environment_variable>
```
Comment (nsoranzo) [inline tools/trinity/tool_dependencies.xml]: By reading https://trinityrnaseq.github.io/performance/mem.html it seems to me it should be safer to also set `--bflyHeapSpaceMax 1G` in tools/trinity/trinity.xml, i.e. `\${TRINITY_MEM_OPTIONS:---max_memory 1G --bflyHeapSpaceMax 1G}`

**Comment by abretaud** [comment]
Thanks for working on this!
Is travis using the new way of resolving all dependencies at once? I think some tests were failing because of this (but maybe I'm confusing with something else...)

**Comment by mvdbeek** [comment]
It is creating mulled environments, yes. Since those are quite long, we're hitting the PaddingError at the moment, but rebuilding the packages seems to be enough to get going. (Inadvertently that's a good way to verify that our dependencies also work when the path is long).

**Comment by nsoranzo** [comment]
@mvdbeek I think I've found the source of the PaddingError issue for `abundance_estimates_to_matrix` and `align_and_estimate_abundance` (but don't know how to solve it): `salmon` requires `icu 56.*` while `r-base 3.3.1` requires `icu 54.*`. Since the older `r-base 3.2.2` does not have a requirement on `icu`, it gets selected when creating the mulled env and causes the PaddingError.

Ping also @bgruening for suggestions.

**Comment by bgruening** [comment]
@mvdbeek @nsoranzo can you try to remove the `r` channel and see if this works? If not, please try to put the `conda-forge` channel as high priority - we are close to remove the `r` channel out of the equation.

**Comment by nsoranzo** [comment]
> @mvdbeek @nsoranzo can you try to remove the r channel and see if this works? If not, please try to put the conda-forge channel as high priority - we are close to remove the r channel out of the equation.

I tried removing the `r` channel, but conda-forge provides only `r-base`, not `r` package. So that fails because `bioconductor-limma` requires `r`.

**Comment by nsoranzo** [comment]
@mvdbeek I've added a commit, but had also to rebase to fix a merge conflict, sorry about that.

**Comment by mvdbeek** [comment]
No problem at all.

**Comment by abretaud** [comment]
Hi,
I just found a bug in the filter_low_expr_transcripts wrapper, do you prefer that I make a PR against your branch @mvdbeek or another IUC PR?

**Comment by mvdbeek** [comment]
You should have write access to this branch, feel free to push directly.

**Comment by abretaud** [comment]
Hum, I don't understand why it's still red, I see no error... Is it normal that there's no planemo test output?

**Comment by nsoranzo** [comment]
@abretaud I'm working on fixing many Conda packages, and I have another pending commit for this PR which also restore the normal logging, I hope to push that today.

**Comment by nsoranzo** [comment]
@abretaud @mvdbeek All Conda packages are updated and working, only the tests for 2 tools fail now, because of some change in Trinity. Can I leave it to you?

**Comment by abretaud** [comment]
Very great! I'll have a look today at the failing tests

**Comment by nsoranzo** [comment]
And also `run_de_analysis` is fixed, thanks @abretaud! Only `analyze_diff_expr` remains.

**Comment by abretaud** [comment]
And it's... green \o/

**Comment by loraine-gueguen** [comment]
@abretaud Thanks!

**Comment by mvdbeek** [comment]
Excellent, thanks a lot @abretaud @nsoranzo!

File: tools/trinity/analyze_diff_expr.xml [hunk 4]
```diff
@@ -94,13 +97,22 @@
...
                 <element name="input.matrix.wt_37_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.0.wt_ph8-UP" compare="sim_size" file="count/analyze_diff_expr/input.matrix.wt_37_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.wt_ph8-UP.subset"/>
-                <element name="input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.0.wt_GSNO-UP" compare="sim_size" file="count/analyze_diff_expr/input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.wt_GSNO-UP.subset"/>
-                <element name="input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.0.wt_ph8-UP" compare="sim_size" file="count/analyze_diff_expr/input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.wt_ph8-UP.subset"/>
+                <!----><element name="input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.0.wt_GSNO-UP" compare="sim_size" file="count/analyze_diff_expr/input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.wt_GSNO-UP.subset"/>
+                <!----><element name="input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.0.wt_ph8-UP" compare="sim_size" file="count/analyze_diff_expr/input.matrix.wt_GSNO_vs_wt_ph8.DESeq2.DE_results.P0.001_C2.wt_ph8-UP.subset"/>
```
Comment (nsoranzo) [inline tools/trinity/analyze_diff_expr.xml]: I think you can remove the 2 `<!---->`, they shouldn't do anything.

Comment (abretaud) [inline tools/trinity/analyze_diff_expr.xml]: Good catch!

**Pull request merged**